src/ng/parse.js

```diff
@@ -1790,7 +1821,7 @@ function $ParseProvider() {
             }
             var lexer = new Lexer($parseOptions);
             var parser = new Parser(lexer, $filter, $parseOptions);
-            parsedExpression = parser.parse(exp);
+            parsedExpression = parser.parse(exp, finalExposeAst);
             parsedExpression.oneTime = !!oneTime;
 
             cache[cacheKey] = addWatchDelegate(parsedExpression);
```

Comment:
One problem is that cached parsedExpressions will always follow the `$$exposeAst` from the first call. E.g.:

```js
$parse('user.name');   // --> No `$$ast` (expected)
$parse({expression: 'user.name', $$exposeAst: true});   // --> No `$$ast` (ooops)
```

Not sure what is the best way to handle that (without impacting uninterested users) 😕 

##
Also, theoretically, the `addInterceptor()` function should preserve the `$$ast` property, but we can assume nobody using `$$exposeAst` cares about interceptors at this point. Maybe at least add a comment somewhere (that adding an interceptor won't preserve the exposed `$$ast` (if any)).